x/mongo/driver/operation.go

```diff
@@ -1546,23 +1540,22 @@ func (op Operation) addClusterTime(dst []byte, desc description.SelectedServer)
 // if the ctx is a Timeout context. If the context is not a Timeout context, it uses the
 // operation's MaxTimeMS if set. If no MaxTimeMS is set on the operation, and context is
 // not a Timeout context, calculateMaxTimeMS returns 0.
-func (op Operation) calculateMaxTimeMS(ctx context.Context, rtt90 time.Duration, rttStats string) (uint64, error) {
+func (op Operation) calculateMaxTimeMS(ctx context.Context, rttMin time.Duration, rttStats string) (uint64, error) {
 	if csot.IsTimeoutContext(ctx) {
 		if deadline, ok := ctx.Deadline(); ok {
 			remainingTimeout := time.Until(deadline)
-			maxTime := remainingTimeout - rtt90
 
 			// Always round up to the next millisecond value so we never truncate the calculated
 			// maxTimeMS value (e.g. 400 microseconds evaluates to 1ms, not 0ms).
-			maxTimeMS := int64((maxTime + (time.Millisecond - 1)) / time.Millisecond)
+			maxTimeMS := remainingTimeout - rttMin
 			if maxTimeMS <= 0 {
 				return 0, fmt.Errorf(
 					"remaining time %v until context deadline is less than or equal to 90th percentile RTT: %w\n%v",
 					remainingTimeout,
 					ErrDeadlineWouldBeExceeded,
 					rttStats)
 			}
-			return uint64(maxTimeMS), nil
+			return uint64(maxTimeMS.Milliseconds()), nil
```

Comment:
Calling `Milliseconds` on a duration value smaller than 1ms will truncate it to 0. We need to retain the "round up" behavior described above to prevent unintentionally sending `maxTimeMS=0` (0 means "no timeout" to the server).